full-service/src/json_rpc/account_secrets.rs

```diff
@@ -22,9 +19,10 @@ pub struct AccountSecrets {
     /// The account ID for this account key in the wallet database.
     pub account_id: String,
 
-    /// The mnemonic from which this account key was derived, as a list of BIP39
-    /// words.
-    pub mnemonic: String,
+    /// The entropy from which this account key was derived, as a String.
+    /// Depending on the version, this will either be a hex encoded string
+    /// for version 1, or a 24 word mnemonic for version 2.
+    pub entropy: String,
```

Comment:
I would suggest leaving `mnemonic` and adding `entropy`, and making each one `Option<String>` - then only populate based on version. 

I would also suggest adding `version` to the AccountSecrets struct